theano/d3viz/d3viz.py

```diff
@@ -78,7 +79,10 @@ def d3viz(fct, outfile, copy_deps=True, *args, **kwargs):
     # Create DOT graph
     formatter = PyDotFormatter(*args, **kwargs)
     graph = formatter(fct)
-    dot_graph = escape_quotes(str(graph.create_dot())).replace('\n', '').replace('\r', '')
+    dot_graph_raw = graph.create_dot()
+    if not six.PY2:
+        dot_graph_raw = dot_graph_raw.decode('utf8')
+    dot_graph = escape_quotes(dot_graph_raw).replace('\n', '').replace('\r', '')
```

Comment:
yes.

On Thu, May 12, 2016 at 2:38 PM, Mikhail Korobov <notifications@github.com>
wrote:

> In theano/d3viz/d3viz.py
> <https://github.com/Theano/Theano/pull/4505#discussion_r63073914>:
>
> > @@ -78,7 +79,10 @@ def d3viz(fct, outfile, copy_deps=True, *args, **kwargs):
> >      # Create DOT graph
> >      formatter = PyDotFormatter(*args, **kwargs)
> >      graph = formatter(fct)
> > -    dot_graph = escape_quotes(str(graph.create_dot())).replace('\n', '').replace('\r', '')
> > +    dot_graph_raw = graph.create_dot()
> > +    if not six.PY2:
> > +        dot_graph_raw = dot_graph_raw.decode('utf8')
> > +    dot_graph = escape_quotes(dot_graph_raw).replace('\n', '').replace('\r', '')
>
> Are you OK with recommending pydot-ng instead of pydot everywhere in docs?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/4505/files/ac2185b20e2dafffdfe93ff2ec63cd37f244dcc4#r63073914>
>